src/output-filedata.c

```diff
@@ -124,6 +124,32 @@ static int CallLoggers(ThreadVars *tv, OutputLoggerThreadStore *store_list,
     return file_logged;
 }
 
+
+/* for hashing truncated files... */
+#ifdef HAVE_NSS
+static void EndHashTruncated( File *ff )
```

Comment:
I don't understand in what case we actually need this. When a file is marked as 'truncated' the same logic is added in this PR, so in what case is this needed?

Reply:
I'm sorry but I don't remember exactly, I recall some flow path leading me to here but probably you're right and I'm wrong so I'm going to remove this in the next PR iteration

Reply:
I think the file state should always be either closed, error or truncated when it is logged, otherwise we have a bug in the file tracking.